src/videojs.ads.js

```diff
@@ -617,19 +617,16 @@ var
             if (!videojs.browser.IS_IOS && player.duration() === Infinity) {
               player.volume(this.preAdVolume_);
             }
-
-            // trigger 'adend' as a consistent notification
-            // event that we're exiting ad-playback.
-            if (player.ads.triggerevent !== 'adend') {
-              player.trigger('adend');
```

Comment:
In a flash environment, where we rely on ExternalInterface communication, when leaving `ad-playback`, this would trigger a second `adend` event. 
For a consistent notification of exiting `ad-playback` state, it is really enough to trigger the `adend` in the eventuality of `adserror` .